dns/dnssec.py

```diff
@@ -239,15 +238,15 @@ def _is_sha512(algorithm):
 
 def _make_hash(algorithm):
     if _is_md5(algorithm):
-        return dns.hash.hashes['MD5']()
+        return MD5.new()
```

Comment:
I could imagine this section rewritten into dict mapping to speedup alg. detection. Can be addressed later

Reply:
I'd generally recommend to rewrite this module to use more declarative codebase and dict mapping detection for algorithms.

It's quite hard to read such long functions and I think due the linear search for used algorithms it may be slower